<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>783</width>
    <height>593</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PyQt5-OpenCV Example</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>threshold upper</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QScrollBar" name="t_max">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>threshold lower</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QScrollBar" name="t_min">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>size max</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QScrollBar" name="s_max">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>150</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>size min</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QScrollBar" name="s_min">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="video">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: 'black'; color: 'white'</string>
        </property>
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="nameText">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QPushButton" name="pushButton_2">
          <property name="text">
           <string>load</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>start recognition</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>783</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_2</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>train_start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>309</x>
     <y>539</y>
    </hint>
    <hint type="destinationlabel">
     <x>392</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>start_recog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>659</x>
     <y>537</y>
    </hint>
    <hint type="destinationlabel">
     <x>526</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>t_max</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>t_max_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>233</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>266</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>t_min</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>t_min_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>614</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>520</x>
     <y>29</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>s_max</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>s_max_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>210</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>227</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>s_min</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>s_min_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>616</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>552</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>train_start()</slot>
  <slot>make_data()</slot>
  <slot>start_recog()</slot>
  <slot>t_max_changed(int)</slot>
  <slot>t_min_changed(int)</slot>
  <slot>s_max_changed(int)</slot>
  <slot>s_min_changed(int)</slot>
 </slots>
</ui>
